addons/sale/models/sale_order.py

```diff
@@ -588,7 +588,7 @@ def create(self, vals_list):
 
     def _compute_field_value(self, field):
         if field.name == 'invoice_status' and not self.env.context.get('mail_activity_automation_skip'):
-            filtered_self = self.filtered(lambda so: (so.user_id or so.partner_id.user_id) and so._origin.invoice_status != 'upselling')
+            filtered_self = self.filtered(lambda so: so.id and (so.user_id or so.partner_id.user_id) and so._origin.invoice_status != 'upselling')
```

Comment:
I seem to recall that newId is truthy anyway?
Don't you have to use "ids" if you want to test that the record is already in database?

This whole fix is a bit light.
There is no real explanation nor any unit testing.
And it's not fixed in the stable that introduces the issues (15 if I follow the linked commit?).

Reply:
Added a generic test in test_mail that generates activities with res_id=0 in transient mode (using a new). No time really for sale specific tests ^^ .